src/common/transformations/src/transformations/symbolic_transformations/symbol_optimization.cpp

```diff
@@ -316,16 +344,16 @@ std::vector<std::shared_ptr<ov::Node>> topological_order(const std::shared_ptr<o
 }
 
 void save_shape_sources(const std::shared_ptr<ov::Node>& op, STS_map& symbol_shape_source) {
-    if (ov::is_type<ov::op::v0::ShapeOf>(op) || ov::is_type<ov::op::v3::ShapeOf>(op)) {
-        const auto& output = op->input_value(0);
+    const auto is_shape_of = ov::is_type<ov::op::util::ShapeOfBase>(op);
+    const auto is_parameter = ov::is_type<ov::op::v0::Parameter>(op);
+    if (is_shape_of || is_parameter) {
+        const auto& output = is_shape_of ? op->input_value(0) : op->output(0);
         if (output.get_partial_shape().rank().is_dynamic())
             return;
         for (const auto& d : output.get_partial_shape()) {
-            if (d.is_static())
-                continue;
-            auto symbol = d.get_symbol();
-            if (symbol == nullptr)
+            if (d.is_static() || d.get_symbol() == nullptr)
```

Comment:
```suggestion
            if (d.is_static() || !d.has_symbol())
```
Maybe `has_symbol` should be updated and static Dimension shouldn't have symbol?